src/crimson/osd/pg_backend.cc

```diff
@@ -60,165 +61,95 @@ PGBackend::PGBackend(shard_id_t shard,
-              auto clone_snap = ss->clone_snaps.find(soid.snap);
-              assert(clone_snap != end(ss->clone_snaps));
-              if (clone_snap->second.empty()) {
-                logger().trace("find_object: {}@[] -- DNE", soid);
-                return crimson::ct_error::enoent::make();
-              }
-              auto first = clone_snap->second.back();
-              auto last = clone_snap->second.front();
-              if (first > soid.snap) {
-                logger().trace("find_object: {}@[{},{}] -- DNE",
-                               soid, first, last);
-                return crimson::ct_error::enoent::make();
-              }
-              logger().trace("find_object: {}@[{},{}] -- HIT",
-                             soid, first, last);
-              return _load_os(soid);
-          });
-        }
-    });
-  }
-}
-
-PGBackend::load_metadata_ertr::future<PGBackend::loaded_object_md_t>
+PGBackend::load_metadata_ertr::future<PGBackend::loaded_object_md_t::ref>
 PGBackend::load_metadata(const hobject_t& oid)
 {
   return store->get_attrs(
     coll,
     ghobject_t{oid, ghobject_t::NO_GEN, shard}).safe_then(
-      [oid, this](auto &&attrs) -> load_metadata_ertr::future<loaded_object_md_t>{
-	loaded_object_md_t ret;
+      [oid, this](auto &&attrs) -> load_metadata_ertr::future<loaded_object_md_t::ref>{
+	loaded_object_md_t::ref ret(new loaded_object_md_t());
 	if (auto oiiter = attrs.find(OI_ATTR); oiiter != attrs.end()) {
 	  bufferlist bl;
 	  bl.push_back(oiiter->second);
-	  ret.os = ObjectState(
+	  ret->os = ObjectState(
 	    object_info_t(bl),
 	    true);
 	} else {
+	  logger().error(
+	    "load_metadata: object {} present but missing object info",
+	    oid);
 	  return crimson::ct_error::object_corrupted::make();
 	}
 	
 	if (oid.is_head()) {
 	  if (auto ssiter = attrs.find(SS_ATTR); ssiter != attrs.end()) {
 	    bufferlist bl;
 	    bl.push_back(ssiter->second);
-	    ret.ss = SnapSet(bl);
+	    ret->ss = SnapSet(bl);
 	  } else {
-	    return crimson::ct_error::object_corrupted::make();
+	    /* TODO: add support for writing out snapsets
+	    logger().error(
+	      "load_metadata: object {} present but missing snapset",
+	      oid);
+	    //return crimson::ct_error::object_corrupted::make();
+	    */
+	    ret->ss = SnapSet();
 	  }
 	}
 
-	return load_metadata_ertr::make_ready_future<loaded_object_md_t>(
+	return load_metadata_ertr::make_ready_future<loaded_object_md_t::ref>(
 	  std::move(ret));
       }, crimson::ct_error::enoent::handle([oid, this] {
-	return load_metadata_ertr::make_ready_future<loaded_object_md_t>(
-	  loaded_object_md_t{
-	    ObjectState(),
-	    std::nullopt
+	logger().debug(
+	  "load_metadata: object {} doesn't exist, returning empty metadata",
+	  oid);
+	return load_metadata_ertr::make_ready_future<loaded_object_md_t::ref>(
+	  new loaded_object_md_t{
+	    ObjectState(
+	      object_info_t(oid),
+	      false
+	    ),
```

Comment:
nit, could you move `),` into the previous line?

Reply:
Ok